src/storage/admin/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,106 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "kvstore/LogEncoder.h"
+#include "storage/admin/RebuildEdgeIndexProcessor.h"
+
+namespace nebula {
+namespace storage {
+
+void RebuildEdgeIndexProcessor::process(const cpp2::RebuildIndexRequest& req) {
+    CHECK_NOTNULL(kvstore_);
+    auto space = req.get_space_id();
+    auto parts = req.get_parts();
+    callingNum_ = parts.size();
+    auto indexID = req.get_index_id();
+    auto itemRet = indexMan_->getEdgeIndex(space, indexID);
+    if (!itemRet.ok()) {
+        cpp2::ResultCode thriftRet;
+        thriftRet.set_code(cpp2::ErrorCode::E_INDEX_NOT_FOUND);
+        codes_.emplace_back(thriftRet);
+        onFinished();
+        return;
+    }
+
+    auto item = itemRet.value();
+    auto edgeType = item->get_schema_id().get_edge_type();
+    LOG(INFO) << "Rebuild Edge Index Space " << space << " Edge Type " << edgeType
+              << " Edge Index " << indexID;
+
+    for (PartitionID part : parts) {
+        std::unique_ptr<kvstore::KVIterator> iter;
+        auto prefix = NebulaKeyUtils::prefix(part);
+        auto ret = kvstore_->prefix(space, part, prefix, &iter);
+        if (ret != kvstore::ResultCode::SUCCEEDED) {
+            LOG(ERROR) << "Processing Part " << part << " Failed";
+            this->pushResultCode(to(ret), part);
+            onFinished();
+            return;
+        }
+
+        auto iterPtr = iter.get();
+        auto atomic = [space, part, edgeType, item, iterPtr, this]() -> std::string {
+            return partitionRebuildIndex(space, part, edgeType, item, iterPtr);
+        };
+
+        auto callback = [space, part, this](kvstore::ResultCode code) {
+            handleAsync(space, part, code);
+        };
+
+        this->kvstore_->asyncAtomicOp(space, part, atomic, callback);
```

Comment:
Keep in mind,  the callback will be invoked in another thread,  and asyncAtomicOp will not block current thread. All variables will be destroyed when the callback accessing it.